### ISSUE TITLE
Can't get `git-branchless --color always smartlog | cat` to show color

### ISSUE THREAD

**Issue by ilyagr** [arxanas/git-branchless#506]
### Description of the bug

I have `git-branchless smartlog` display pretty colors as expected, but `git-branchless --color always smartlog | cat` does not show colors.

Neither does `git -c color.ui=always sl |cat`. ( `git -c color.ui=always log | head` is the way to force normal `git log` to show colors).

Additionally (slightly separate bug), `git branchless smartlog --color=always` and `git sl --color=always` result in an error and exit code 2:

```
error: Found argument '--color' which wasn't expected, or isn't valid in this context

        If you tried to supply `--color` as a value rather than a flag, use `-- --color`

USAGE:
    git-branchless smartlog [OPTIONS] [REVSET]

For more information try --help
```

(My real goal is to get `watch --color git-branchless smartlog` to work with color).

#### Software version

git-branchless 0.4.0

#### Operating system

Linux 5.10.120-16471-g167ad2e5dcba

#### Command-line

```bash
/home/ilyagr/.local/app/cargo/bin/git-branchless bug-report 
```

#### Environment variables

```bash
SHELL=/usr/bin/fish
EDITOR=vim
```

#### Git version

```
> git version 
git version 2.30.2
```

#### Hooks

<details>
<summary>Show 6 hooks</summary>

##### Hook `post-commit`

```
#!/bin/sh
## START BRANCHLESS CONFIG

git branchless hook-post-commit "$@"
## END BRANCHLESS CONFIG
```
##### Hook `post-merge`

```
#!/bin/sh
## START BRANCHLESS CONFIG

git branchless hook-post-merge "$@"
## END BRANCHLESS CONFIG
```
##### Hook `post-rewrite`

```
#!/bin/sh
## START BRANCHLESS CONFIG

git branchless hook-post-rewrite "$@"
## END BRANCHLESS CONFIG
```
##### Hook `post-checkout`

```
#!/bin/sh
## START BRANCHLESS CONFIG

git branchless hook-post-checkout "$@"
## END BRANCHLESS CONFIG
```
##### Hook `pre-auto-gc`

```
#!/bin/sh
## START BRANCHLESS CONFIG

git branchless hook-pre-auto-gc "$@"
## END BRANCHLESS CONFIG
```
##### Hook `reference-transaction`

```
#!/bin/sh
## START BRANCHLESS CONFIG

# Avoid canceling the reference transaction in the case that `branchless` fails
# for whatever reason.
git branchless hook-reference-transaction "$@" || (
echo 'branchless: Failed to process reference transaction!'
echo 'branchless: Some events (e.g. branch updates) may have been lost.'
echo 'branchless: This is a bug. Please report it.'
)
## END BRANCHLESS CONFIG
```

</details>

#### Events

<details>
<summary>Show 5 events</summary>

##### Event ID: 4, transaction ID: 24 (message: branchstack)

1. `RefUpdateEvent { timestamp: 1661039030.7608836, event_tx_id: EventTransactionId(24), ref_name: "refs/heads/redacted-ref-0", old_oid: df32947b3858c86debaf2140b00429692f6f7dea, new_oid: f055b5f8ffd7c2b4090ab02603a016a73bb93bb0, message: None }`
1. `RefUpdateEvent { timestamp: 1661039030.960924, event_tx_id: EventTransactionId(24), ref_name: "refs/heads/redacted-ref-1", old_oid: 45175bf1adf6a71433e99117f62f24cd10d93f1b, new_oid: 3323b0133b1814cf44b4487b51b9a173cc25351c, message: None }`
```
:
O 47468a8 6d xxxxx xxxx xxxxxxx xxxx xxxx xxxxxxxxxxx
|\
| o ea9136b 4d xxxxx xxx xxxxx xxx
| |
| o 6702ad4 4d xxxx xxx xxxxxxx
| |
| o 06291f0 20h xxx xxxxxx
| |
| o 609f430 20h (redacted-ref-2) xxx x xxxx xxxxx xxx xxxxxxx xx xxx xxxxxx
|\
| o 68ed6c8 18h xxxxxxxxxx xxxxx xxxxx
| |
| o 960764a 18h (redacted-ref-3) xxxxxxxxxxxxxx xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|
O 3009e33 5d (redacted-ref-4, remote origin/master) xxxxx xxxx xxxxxxx xxxx xxxx xxxxxxxxxxxxxxxxxxxxxxxxxxxx
|\
| o 3323b01 4m (redacted-ref-1) xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|\
| o 7848726 4m xxxxxxxxxx xxxxx xxxxx
| |
| @ e005885 4m (redacted-ref-5) xxxxxxxxxxxxxx xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|
o f055b5f 4m (redacted-ref-0) xxxxx xxxxx
```
##### Event ID: 2, transaction ID: 20 (message: revise)

1. `RefUpdateEvent { timestamp: 1661039004.5472133, event_tx_id: EventTransactionId(20), ref_name: "refs/heads/redacted-ref-5", old_oid: 98e6f1fa080725aeb6625a23cd7435a556587636, new_oid: e005885fe942fcdf29cf9820547e22e4c5b0720c, message: None }`
1. `RefUpdateEvent { timestamp: 1661039004.5472133, event_tx_id: EventTransactionId(20), ref_name: "HEAD", old_oid: 98e6f1fa080725aeb6625a23cd7435a556587636, new_oid: e005885fe942fcdf29cf9820547e22e4c5b0720c, message: None }`
```
:
O 47468a8 6d xxxxx xxxx xxxxxxx xxxx xxxx xxxxxxxxxxx
|\
| o ea9136b 4d xxxxx xxx xxxxx xxx
| |
| o 6702ad4 4d xxxx xxx xxxxxxx
| |
| o 06291f0 20h xxx xxxxxx
| |
| o 609f430 20h (redacted-ref-2) xxx x xxxx xxxxx xxx xxxxxxx xx xxx xxxxxx
|\
| o 68ed6c8 18h xxxxxxxxxx xxxxx xxxxx
| |
| o 960764a 18h (redacted-ref-3) xxxxxxxxxxxxxx xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|
O 3009e33 5d (redacted-ref-4, remote origin/master) xxxxx xxxx xxxxxxx xxxx xxxx xxxxxxxxxxxxxxxxxxxxxxxxxxxx
|\
| o 3323b01 4m (redacted-ref-1) xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|\
| o 7848726 4m xxxxxxxxxx xxxxx xxxxx
| |
| @ e005885 4m (redacted-ref-5) xxxxxxxxxxxxxx xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|
o f055b5f 4m (redacted-ref-0) xxxxx xxxxx
```
##### Event ID: 1, transaction ID: 14 (message: reference-transaction)

1. `RefUpdateEvent { timestamp: 1661038903.3202953, event_tx_id: EventTransactionId(14), ref_name: "refs/heads/redacted-ref-0", old_oid: 5e1c636bb6ff388fb54ec105dd41bac0894caaad, new_oid: df32947b3858c86debaf2140b00429692f6f7dea, message: None }`
```
:
O 47468a8 6d xxxxx xxxx xxxxxxx xxxx xxxx xxxxxxxxxxx
|\
| o ea9136b 4d xxxxx xxx xxxxx xxx
| |
| o 6702ad4 4d xxxx xxx xxxxxxx
| |
| o 06291f0 20h xxx xxxxxx
| |
| o 609f430 20h (redacted-ref-2) xxx x xxxx xxxxx xxx xxxxxxx xx xxx xxxxxx
|\
| o 68ed6c8 18h xxxxxxxxxx xxxxx xxxxx
| |
| o 960764a 18h (redacted-ref-3) xxxxxxxxxxxxxx xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|
O 3009e33 5d (redacted-ref-4, remote origin/master) xxxxx xxxx xxxxxxx xxxx xxxx xxxxxxxxxxxxxxxxxxxxxxxxxxxx
|\
| o 3323b01 4m (redacted-ref-1) xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|\
| o 7848726 4m xxxxxxxxxx xxxxx xxxxx
| |
| @ e005885 4m (redacted-ref-5) xxxxxxxxxxxxxx xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|
o f055b5f 4m (redacted-ref-0) xxxxx xxxxx
```
There are no previous available events.
```
:
O 47468a8 6d xxxxx xxxx xxxxxxx xxxx xxxx xxxxxxxxxxx
|\
| o ea9136b 4d xxxxx xxx xxxxx xxx
| |
| o 6702ad4 4d xxxx xxx xxxxxxx
| |
| o 06291f0 20h xxx xxxxxx
| |
| o 609f430 20h (redacted-ref-2) xxx x xxxx xxxxx xxx xxxxxxx xx xxx xxxxxx
|\
| o 68ed6c8 18h xxxxxxxxxx xxxxx xxxxx
| |
| o 960764a 18h (redacted-ref-3) xxxxxxxxxxxxxx xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|
O 3009e33 5d (redacted-ref-4, remote origin/master) xxxxx xxxx xxxxxxx xxxx xxxx xxxxxxxxxxxxxxxxxxxxxxxxxxxx
|\
| o 3323b01 4m (redacted-ref-1) xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|\
| o 7848726 4m xxxxxxxxxx xxxxx xxxxx
| |
| @ e005885 4m (redacted-ref-5) xxxxxxxxxxxxxx xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|
o f055b5f 4m (redacted-ref-0) xxxxx xxxxx
```
There are no previous available events.
```
:
O 47468a8 6d xxxxx xxxx xxxxxxx xxxx xxxx xxxxxxxxxxx
|\
| o ea9136b 4d xxxxx xxx xxxxx xxx
| |
| o 6702ad4 4d xxxx xxx xxxxxxx
| |
| o 06291f0 20h xxx xxxxxx
| |
| o 609f430 20h (redacted-ref-2) xxx x xxxx xxxxx xxx xxxxxxx xx xxx xxxxxx
|\
| o 68ed6c8 18h xxxxxxxxxx xxxxx xxxxx
| |
| o 960764a 18h (redacted-ref-3) xxxxxxxxxxxxxx xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|
O 3009e33 5d (redacted-ref-4, remote origin/master) xxxxx xxxx xxxxxxx xxxx xxxx xxxxxxxxxxxxxxxxxxxxxxxxxxxx
|\
| o 3323b01 4m (redacted-ref-1) xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|\
| o 7848726 4m xxxxxxxxxx xxxxx xxxxx
| |
| @ e005885 4m (redacted-ref-5) xxxxxxxxxxxxxx xxx xxxx xxxxx xxx xxxxxxx xxxx xxx xxxxxx
|
o f055b5f 4m (redacted-ref-0) xxxxx xxxxx
```

</details>

**Comment by ilyagr** [comment]
One possible resolution would be to remove the `--color` option from git-branchless entirely and just use git's `color.ui` option. In this case, it would be nice to also add support for `git-branchless -c color.ui=always`, so that it works the same as `git -c color.ui=always branchless`. This could work for other options as well.

**Comment by ilyagr** [comment]
https://github.com/gitext-rs/git-stack/issues/17 may be relevant for the approach in my previous comment. Does git-branchless support `git -c` configuration at all?

**Comment by arxanas** [comment]
Hi @ilyagr, thanks for reporting! I was able to reproduce this issue. Can you try running with the fix in #512 to confirm that it fixes your issue? (See [Manual testing](https://github.com/arxanas/git-branchless/wiki/Runbook#manual-testing).)

> Does git-branchless support git -c configuration at all?

No, git-branchless currently doesn't support anything like that. Feel free to open a pull request to implement that behavior, as it seems reasonable. I guess we would want to include user configuration in `Repo::get_readonly_config`, and then maybe we'd also have to ensure that Git subprocesses inherit that configuration?

> (My real goal is to get watch --color git-branchless smartlog to work with color).

In the long term, git-branchless would have an interactive interface which supports this workflow. In that case, would you prefer a terminal-based or graphical solution for the interface?

**Comment by ilyagr** [comment]
It all seems to work fine now. Thank you very much for the quick fix! Thanks also for the easy-to-follow instructions on manual testing!

> In the long term, git-branchless would have an interactive interface which supports this workflow. In that case, would you prefer a terminal-based or graphical solution for the interface?

Terminal-based would definitely be my preference. I often work via SSH.

A GUI might make more sense as an integration with VS Code or some other GUI tool. A `gitk`-like GUI would be cute, but I personally don't know if I'd ever use it.
